### PR TITLE
Prerequisite for Discovered Apps

### PR DESCRIPTION
**Pull request by prsurve**

### CODE REVIEW

File: ocs_ci/deployment/deployment.py [hunk 1]
```diff
@@ -3467,6 +3497,10 @@ def deploy(self):
         for i in acm_indexes:
             config.switch_ctx(i)
             self.create_dpa(self.meta_obj.bucket_name)
+
+        config.switch_acm_ctx()
+        # Adding Ca Cert
+        self.add_cacert_ramen_configmap()
```
Comment (Akarsha-rai) [inline ocs_ci/deployment/deployment.py:3503]: Isn't the creation of dpa is missing?

File: ocs_ci/deployment/deployment.py [hunk 2]
```diff
@@ -3411,7 +3431,17 @@ def deploy(self):
         for i in acm_indexes:
             config.switch_ctx(i)
             self.create_dpa(self.meta_obj.bucket_name)
+
```
Comment (Akarsha-rai) [inline ocs_ci/deployment/deployment.py:3434]: Is OADP operator installation is covered here?

File: ocs_ci/deployment/deployment.py [hunk 3]
```diff
@@ -3411,7 +3431,17 @@ def deploy(self):
...
+        config.switch_acm_ctx()
+        # Adding Ca Cert
+        self.add_cacert_ramen_configmap()
         # Only on the active hub enable managedserviceaccount-preview
+        managed_clusters = get_non_acm_cluster_config()
+        for cluster in managed_clusters:
+            index = cluster.MULTICLUSTER["multicluster_index"]
+            config.switch_ctx(index)
+            logger.info("Creating Resource DataProtectionApplication")
+            run_cmd(f"oc create -f {constants.DPA_DISCOVERED_APPS_PATH}")
```
Comment (Akarsha-rai) [inline ocs_ci/deployment/deployment.py:3444]: Is it good to add the DR secrets are created or not as per the doc steps?

Comment (prsurve) [reply ocs_ci/deployment/deployment.py:3444]: We don't have to create secret as it's automated process

File: ocs_ci/deployment/deployment.py (hunk 2 above)
Comment (prsurve) [reply ocs_ci/deployment/deployment.py:3434]: Yes it's covered

**Review by Akarsha-rai — DISMISSED** [review]
lgtm

**Review by Akarsha-rai — APPROVED** [review]

**Review by am-agrawa — APPROVED** [review]

**Review by dahorak — APPROVED** [review]

**Pull request merged**